Example-iOS-Swift/WebViewController.swift

```diff
@@ -17,15 +18,54 @@ class WebViewController: UIViewController, WKScriptMessageHandler {
     var avatarUrlDelegate:WebViewDelegate?
     var webView: WKWebView!
     let cookieName = "rpm-uid"
+    var subscriptionCreated = false
     
     //Update to your custom URL here
-    let readyPlayerMeUrl = URL(string: "https://demo.readyplayer.me/avatar")!
+    //let readyPlayerMeUrl = URL(string: "https://demo.readyplayer.me/avatar")!
+    let readyPlayerMeUrl = URL(string: "https://demo.develop.readyplayer.me/avatar?frameApi")!
     
-    let source = "window.addEventListener('message', function(event){ document.querySelector('.content').remove(); setTimeout(() => {window.webkit.messageHandlers.iosListener.postMessage(event.data);}, 1000) });"
+        let source = """
+            window.addEventListener('message', function(event){
+                const json = parse(event)
+        
+                if(event.data.includes('.glb')){
+                    window.webkit.messageHandlers.iosListener.postMessage(event.data);
+                    return;
+                }
+
+                if (json?.source !== 'readyplayerme') {
+                  return;
+                }
+
+                // Susbribe to all events sent from Ready Player Me once frame is ready
+                if (json.eventName === 'v1.frame.ready') {
+                  window.postMessage(
+                    JSON.stringify({
+                      target: 'readyplayerme',
+                      type: 'subscribe',
+                      eventName: 'v1.**'
+                    }),
+                    '*'
+                  );
+                }
+
+                window.webkit.messageHandlers.iosListener.postMessage(event.data);
+
+
+                function parse(event) {
+                    try {
+                        return JSON.parse(event.data)
+                    } catch (error) {
+                        return null
+                    }
+                };
+            });
+        """
     
     override func loadView(){
+        
         let config = WKWebViewConfiguration()
-        let script = WKUserScript(source: source, injectionTime: .atDocumentEnd, forMainFrameOnly: false)
+        let script = WKUserScript(source: source, injectionTime: .atDocumentStart, forMainFrameOnly: false)
```

Comment:
Ah this is what you told me about.

Reply:
yup! not sure if it makes any difference in swift example though